### PR TITLE
Added multi-channel devices and get/set power state

### PR DESCRIPTION
**Pull request by NickDub**

### CODE REVIEW

**Comment by skydiver** [comment]
Hey @NickDub 

Thanks for the PR, looks great.

Did a few small fixes, but i don't have a multichannel device to test; can you do a final test?


Thanks.-

**Comment by skydiver** [comment]
Hey @NickDub 

Thanks for the PR, looks great.

Did a few small fixes, but i don't have a multichannel device to test; can you do a final test?


Thanks.-

**Comment by NickDub** [comment]
Hi @skydiver ,

Thanks for your quick feedback :wink:

Sorry, but I don't have a multichannel device to test, but for the code, I was inspired by these 2 projects:
- http://pgeorgiev.com/how-do-sonoff-devices-work/
- https://github.com/howanghk/homebridge-ewelink

I also added the following features: 
- Authentication error handling.
- Automatic region changing, if it is not good (I'm in the 'eu' :yum:).
- Channels number controlling, according to the UIID.

Bluesky (I'm skydiver too)

File: main.js [hunk 1]
```diff
@@ -107,6 +171,32 @@ class eWeLink {
 
     return { status: 'ok', state };
   }
+
+  async toggleDevice(deviceId, channel = 1) {
+    const powerState = await this.getDevicePowerState(deviceId, channel);
+    const state = _get(powerState, 'state', false);
+
+    if (!state) {
+      return { error: powerState.error, msg: device.msg | 'Device does not exist' };
```
Comment (skydiver) [inline main.js]: device doesn't exists here
will return just the `Device does not exist` string

File: main.js [hunk 2]
```diff
@@ -107,6 +171,32 @@ class eWeLink {
...
+      newState,
+      channel
+    );
+
+    return newResponse;
+  }
+
+  async getDeviceChannelCount(deviceId) {
+    const device = await this.getDevice(deviceId);
+    const switchesAmount = getDeviceChannelCount(device);
```
Comment (skydiver) [inline main.js]: i've renamed this method to not get confused by this and the one on helpers file

**Comment by skydiver** [comment]
@NickDub 

new additions looks great, love the login/region error retry 👏 👏 

i did a very small changes, mostly related to linting

what do you think about merge this one?

thanks!

**Comment by NickDub** [comment]
@skydiver 

Nice job 👍 
It's ok for me to merge!

Thanks

**Comment by skydiver** [comment]
Thanks for the contribution @NickDub 
Version 1.2.0 was published to npm!

**Pull request merged**